Update to 4.26 and product name change

romanCattaneo opened the pull request:
## Description

@RalucaNicola This PR updates the samples to the latest API version and reflects the name change of the product from js api to sdk for js.

Some more details on other things in this PR

- Reworded top-level README and added a screenshot of the website for people to see your impressive work at first a glance.
- Updated top-level `package.json` - not really sure what it's needed for, but it had some outdated and conflicting information in there (e.g. license field was set to `MIT`, whereas everywhere it says that this work is licensed under Apache 2.0)
- I created issues for things that were out of scope for a simple regression check after the update. There are some places where data is not available (anymore), see issue https://github.com/RalucaNicola/JSAPI_demos/issues/30.
- One or two samples also couldn't be updated. I marked the inline and created issues.

Commit:
Update README (name change)

Updated the README to reflect the name change of the js api product.
Also re-worded and added a screenshot of the website for people to see
your impressive work at a glance.

Commit:
Update top-level package.json

- Reflect the name change
- Change license from MIT to Apache-2.0, which is distributed with the
  code and advertised as such on GH.

ghost commented:
Also updated `package.lock` file

- reflect name change
- change license from `MIT` to `Apache-2.0`, which is shipped in the `LICENSE` file in the code and advertised as such on Github
- updated lockfile v1 -> v2

Thinking about this a bit, why is there a top-level `package.json` in the first place?

Commit:
Update index.html with product name change

Commit:
Update samples to 4.26

This updates samples to js api version 4.26 and reflects the name change
from api to sdk in all of them.

Created issues for samples, which can't be updates and for those who
have problems and/or show deprecation warnings.

romanCattaneo commented on art-galleries-nyc/main.js:
```diff
@@ -19,7 +19,6 @@ require([
   "esri/symbols/TextSymbol3DLayer",
 
   "esri/widgets/Search",
-  "dojo/domReady!"
```
Many of these `dojo/` things remaining. They look like memories from an old past and are not necessary anymore. With 4.26 we cleaned up a bit in the API and thus these files are not served anymore. In all the cases in this PR, they were not actually needed for the sample to work. So... less code - yay!

romanCattaneo commented on cities-globe/README.md:
````diff
@@ -294,9 +293,11 @@ const populationLayer = new FeatureLayer({
 // don't forget to add it to the web scene
 webscene.addMany([countryBoundaries, populationLayer]);
 ```
+
````
Adding a blank line around fenced code is best practice

romanCattaneo commented on low-poly-world/index.html:
```diff
@@ -4,7 +4,7 @@
     <meta name="viewport" content="initial-scale=1" />
     <title>Low poly map of the world</title>
-    <link rel="stylesheet" href="https://js.arcgis.com/4.24/esri/css/main.css" />
+    <link rel="stylesheet" href="https://js.arcgis.com/4.23/esri/css/main.css" />
```
This one I actually downgraded because it broke with 4.24 from a breaking change, see issue https://github.com/RalucaNicola/JSAPI_demos/issues/21.

romanCattaneo commented on package.json:
```diff
@@ -12,7 +12,7 @@
   ],
   "author": "Raluca Nicola",
-  "license": "MIT",
+  "license": "Apache-2.0",
```
License change

romanCattaneo commented on sanfrancisco-buildings/index.html:
```diff
@@ -6,7 +6,7 @@
     <meta name="viewport" content="initial-scale=1,maximum-scale=1,user-scalable=no" />
     <title>San Francisco buildings</title>
-    <link rel="stylesheet" href="https://js.arcgis.com/4.24/esri/css/main.css" />
+    <link rel="stylesheet" href="https://js.arcgis.com/4.25/esri/css/main.css" />
```
I couldn't update SF buildings all the way to 4.26 because of some weird editing issue, see issue https://github.com/RalucaNicola/JSAPI_demos/issues/20. Didn't have time to check if this is because of your app or because of a regression in the API. I'll report this to the experience team.

romanCattaneo reviewed (commented):
Some remarks inline

ghost commented:
Ping - friendly reminder to review (& merge)

RalucaNicola reviewed (approved):
Thanks so much for the upgrade to the new version!
